include/intersection_point.h

```diff
@@ -9,10 +9,12 @@ class IntersectionPoint {
   Vertex position_;
   Direction normal_;
   Material material_;
+  float z_;
```

Comment:
I think it's wrong to use z. And probably wrong to use t too.
Think just a plain "distance_" is enough.. or "distance_to_origin_" if you wanna be more accurate